File: iosp1/P1/P1/UI/Screens/WelcomeScreen.swift
import SwiftUI

struct WelcomeScreen: View {
    let onStartTracking: () -> Void
    let onGoToHome: () -> Void
    
    @State private var currentSlide = 0
    
    private let slides: [(String, String)] = [
        (String(localized: "welcome_slide_1_title"), String(localized: "welcome_slide_1_body")),
        (String(localized: "welcome_slide_2_title"), String(localized: "welcome_slide_2_body")),
        (String(localized: "welcome_slide_3_title"), String(localized: "welcome_slide_3_body"))
    ]
    
    var body: some View {
        VStack(spacing: 0) {
            Spacer().frame(height: 60)
            
            Image("logo")
                .resizable()
                .scaledToFit()
                .frame(width: 80, height: 80)
                .clipShape(RoundedRectangle(cornerRadius: 16))
            
            Spacer()
            
            VStack(spacing: 16) {
                Text(slides[currentSlide].0)
                    .font(.system(size: 32, weight: .heavy, design: .default))
                    .multilineTextAlignment(.center)
                    .padding(.horizontal, 20)
                
                Text(slides[currentSlide].1)
                    .font(.body)
                    .foregroundColor(Color(UIColor.systemGray))
                    .multilineTextAlignment(.center)
                    .padding(.horizontal, 24)
                    .lineSpacing(6)
            }
            .id(currentSlide)
            .transition(.asymmetric(
                insertion: .opacity.combined(with: .move(edge: .trailing)),
                removal: .opacity.combined(with: .move(edge: .leading))
            ))
            
            Spacer()
            
            HStack(spacing: 8) {
                ForEach(0..<slides.count, id: \.self) { index in
                    Circle()
                        .fill(currentSlide == index ? Color.black : Color(UIColor.systemGray5))
                        .frame(width: currentSlide == index ? 10 : 8, height: currentSlide == index ? 10 : 8)
                        .animation(.spring(response: 0.3, dampingFraction: 0.7), value: currentSlide)
                }
            }
            .padding(.vertical, 32)
            
            VStack(spacing: 16) {
                Link("By continuing, you accept our Terms of Use & Privacy Policy", destination: URL(string: "https://p1-privacy-policy.pages.dev/")!)
                    .font(.system(size: 11))
                    .foregroundColor(Color(UIColor.systemGray))
                    .underline()
                    .padding(.bottom, 8)
                    
                if currentSlide < slides.count - 1 {
                    LpmPrimaryButton(text: String(localized: "welcome_next")) {
                        withAnimation {
                            currentSlide += 1
                        }
                    }
                    
                    Button(action: onGoToHome) {
                        Text(String(localized: "welcome_skip"))
                            .fontWeight(.medium)
                            .foregroundColor(Color(UIColor.systemGray))
                    }
                    .padding(.vertical, 8)
                } else {
                    LpmPrimaryButton(text: String(localized: "welcome_start"), action: onStartTracking)
                    
                    Button(action: onGoToHome) {
                        Text(String(localized: "welcome_go_home"))
                            .fontWeight(.medium)
                            .foregroundColor(Color(UIColor.systemGray))
                    }
                    .padding(.vertical, 8)
                }
            }
            .padding(.horizontal, 28)
            .padding(.bottom, 12)

            Text("v\(Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String ?? "1.0")")
                .font(.system(size: 11))
                .foregroundColor(Color(UIColor.systemGray3))
                .padding(.bottom, 12)
        }
        .background(Color.white)
    }
}

#Preview {
    WelcomeScreen(onStartTracking: {}, onGoToHome: {})
}

File: androidp1/app/src/main/java/com/preappointment1/app/ui/screens/WelcomeScreen.kt
package com.preappointment1.app.ui.screens

import androidx.compose.animation.*
import androidx.compose.animation.core.tween
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.compose.ui.graphics.Color
import com.preappointment1.app.R
import com.preappointment1.app.ui.components.LpmPrimaryButton
import com.preappointment1.app.ui.theme.*

@Composable
fun WelcomeScreen(
    onStartTracking: () -> Unit,
    onGoToHome: () -> Unit,
    modifier: Modifier = Modifier
) {
    var currentSlide by remember { mutableIntStateOf(0) }

    val slides = listOf(
        Pair(
            stringResource(R.string.welcome_slide_1_title),
            stringResource(R.string.welcome_slide_1_body)
        ),
        Pair(
            stringResource(R.string.welcome_slide_2_title),
            stringResource(R.string.welcome_slide_2_body)
        ),
        Pair(
            stringResource(R.string.welcome_slide_3_title),
            stringResource(R.string.welcome_slide_3_body)
        )
    )

    Scaffold(
        containerColor = CanvasBackground,
        modifier = modifier
    ) { padding ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(padding)
                .padding(horizontal = 28.dp, vertical = 40.dp),
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.SpaceBetween
        ) {
            Spacer(modifier = Modifier.weight(1f))

            AnimatedContent(
                targetState = currentSlide,
                transitionSpec = {
                    fadeIn(animationSpec = tween(400)) togetherWith fadeOut(animationSpec = tween(400))
                },
                label = "slide_transition"
            ) { targetSlide ->
                Column(
                    horizontalAlignment = Alignment.CenterHorizontally,
                    modifier = Modifier.fillMaxWidth()
                ) {
                    Text(
                        text = slides[targetSlide].first,
                        style = MaterialTheme.typography.headlineLarge,
                        fontWeight = FontWeight.ExtraBold,
                        textAlign = TextAlign.Center,
                        color = TextPrimary,
                        lineHeight = 38.sp,
                        modifier = Modifier.padding(bottom = 16.dp)
                    )
                    Text(
                        text = slides[targetSlide].second,
                        style = MaterialTheme.typography.bodyLarge,
                        color = TextSecondary,
                        textAlign = TextAlign.Center,
                        lineHeight = 24.sp,
                        modifier = Modifier.padding(horizontal = 16.dp)
                    )
                }
            }

            Spacer(modifier = Modifier.weight(1f))

            Row(
                modifier = Modifier.padding(vertical = 32.dp),
                horizontalArrangement = Arrangement.spacedBy(8.dp),
                verticalAlignment = Alignment.CenterVertically
            ) {
                repeat(slides.size) { index ->
                    Box(
                        modifier = Modifier
                            .size(if (currentSlide == index) 12.dp else 8.dp)
                            .clip(CircleShape)
                            .background(if (currentSlide == index) SagePrimary else MintBadge)
                    )
                }
            }

            Column(
                modifier = Modifier.fillMaxWidth(),
                verticalArrangement = Arrangement.spacedBy(16.dp),
                horizontalAlignment = Alignment.CenterHorizontally
            ) {
                val uriHandler = androidx.compose.ui.platform.LocalUriHandler.current
                Text(
                    text = "By continuing, you accept our Terms of Use & Privacy Policy",
                    fontSize = 12.sp,
                    color = TextMuted,
                    textDecoration = androidx.compose.ui.text.style.TextDecoration.Underline,
                    textAlign = TextAlign.Center,
                    modifier = Modifier
                        .padding(bottom = 8.dp)
                        .clickable { uriHandler.openUri("https://p1-privacy-policy.pages.dev/") }
                )

                if (currentSlide < slides.size - 1) {
                    LpmPrimaryButton(
                        text = stringResource(R.string.welcome_next),
                        onClick = { currentSlide++ }
                    )
                    TextButton(
                        onClick = onGoToHome,
                        modifier = Modifier.fillMaxWidth()
                    ) {
                        Text(
                            stringResource(R.string.welcome_skip),
                            color = TextSecondary,
                            fontWeight = FontWeight.SemiBold
                        )
                    }
                } else {
                    LpmPrimaryButton(
                        text = stringResource(R.string.welcome_start),
                        onClick = onStartTracking
                    )
                    TextButton(
                        onClick = onGoToHome,
                        modifier = Modifier.fillMaxWidth()
                    ) {
                        Text(
                            stringResource(R.string.welcome_go_home),
                            color = TextSecondary,
                            fontWeight = FontWeight.SemiBold
                        )
                    }
                }
            }
        }
    }
}
